File: androidApp/src/main/kotlin/com/weather/android/ui/templates/AtomicComponentShowcase.kt
package com.weather.android.ui.templates

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.weather.android.ui.atoms.button.PrimaryButton
import com.weather.android.ui.atoms.button.SecondaryButton
import com.weather.android.ui.atoms.card.AtomicCard
import com.weather.android.ui.atoms.card.AtomicElevatedCard
import com.weather.android.ui.atoms.card.AtomicOutlinedCard
import com.weather.android.ui.atoms.input.AtomicCheckbox
import com.weather.android.ui.atoms.input.AtomicSearchField
import com.weather.android.ui.atoms.input.AtomicSwitch
import com.weather.android.ui.atoms.input.AtomicTextField
import com.weather.android.ui.atoms.navigation.AtomicBottomBar
import com.weather.android.ui.atoms.navigation.AtomicBottomBarItem
import com.weather.android.ui.atoms.navigation.AtomicTopBarWithBack
import com.weather.android.ui.atoms.navigation.BottomNavItem
import com.weather.android.ui.molecules.form.SearchForm
import com.weather.android.ui.molecules.form.SettingsForm
import com.weather.android.ui.molecules.form.SettingsFormData
import com.weather.android.ui.theme.AtomicDesignSystem
import com.weather.android.ui.theme.AtomicTheme

/**
 * Showcase template demonstrating all atomic components and molecules
 * 
 * This template serves as:
 * - Documentation of available components
 * - Visual testing of component integration
 * - Design system validation
 * - Component behavior demonstration
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AtomicComponentShowcase(
    modifier: Modifier = Modifier
) {
    var searchQuery by remember { mutableStateOf("") }
    var textFieldValue by remember { mutableStateOf("") }
    var checkboxChecked by remember { mutableStateOf(false) }
    var switchChecked by remember { mutableStateOf(true) }
    var settingsData by remember { mutableStateOf(SettingsFormData()) }
    var selectedBottomNavItem by remember { mutableStateOf("home") }
    
    val bottomNavItems = listOf(
        BottomNavItem("Home", Icons.Default.Home, "home"),
        BottomNavItem("Favorites", Icons.Default.Favorite, "favorites"),
        BottomNavItem("Settings", Icons.Default.Settings, "settings")
    )
    
    Scaffold(
        topBar = {
            AtomicTopBarWithBack(
                title = "Atomic Components",
                onBackClick = { /* Back navigation */ },
                centered = true
            )
        },
        bottomBar = {
            AtomicBottomBar {
                bottomNavItems.forEach { item ->
                    AtomicBottomBarItem(
                        selected = selectedBottomNavItem == item.route,
                        onClick = { selectedBottomNavItem = item.route },
                        icon = {
                            Icon(
                                imageVector = item.icon,
                                contentDescription = item.label
                            )
                        },
                        label = {
                            Text(
                                text = item.label,
                                style = AtomicDesignSystem.typography.LabelMedium
                            )
                        }
                    )
                }
            }
        }
    ) { paddingValues ->
        Column(
            modifier = modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(AtomicDesignSystem.spacing.MD)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(AtomicDesignSystem.spacing.MD)
        ) {
            // Input Atoms Section
            Text(
                text = "Input Atoms",
                style = AtomicDesignSystem.typography.HeadlineSmall
            )
            
            AtomicTextField(
                value = textFieldValue,
                onValueChange = { textFieldValue = it },
                label = "Text Field",
                placeholder = "Enter some text..."
            )
            
            AtomicSearchField(
                value = searchQuery,
                onValueChange = { searchQuery = it },
                placeholder = "Search locations..."
            )
            
            AtomicCheckbox(
                checked = checkboxChecked,
                onCheckedChange = { checkboxChecked = it },
                label = "Enable notifications"
            )
            
            AtomicSwitch(
                checked = switchChecked,
                onCheckedChange = { switchChecked = it },
                label = "Dark mode"
            )
            
            Spacer(modifier = Modifier.height(AtomicDesignSystem.spacing.SM))
            
            // Button Atoms Section
            Text(
                text = "Button Atoms",
                style = AtomicDesignSystem.typography.HeadlineSmall
            )
            
            PrimaryButton(
                text = "Primary Action",
                onClick = { /* Primary action */ }
            )
            
            SecondaryButton(
                text = "Secondary Action",
                onClick = { /* Secondary action */ }
            )
            
            Spacer(modifier = Modifier.height(AtomicDesignSystem.spacing.SM))
            
            // Card Atoms Section
            Text(
                text = "Card Atoms",
                style = AtomicDesignSystem.typography.HeadlineSmall
            )
            
            AtomicCard(
                onClick = { /* Card click */ }
            ) {
                Text(
                    text = "Basic Card",
                    modifier = Modifier.padding(AtomicDesignSystem.spacing.MD),
                    style = AtomicDesignSystem.typography.BodyLarge
                )
            }
            
            AtomicOutlinedCard {
                Text(
                    text = "Outlined Card",
                    modifier = Modifier.padding(AtomicDesignSystem.spacing.MD),
                    style = AtomicDesignSystem.typography.BodyLarge
                )
            }
            
            AtomicElevatedCard(
                onClick = { /* Elevated card click */ }
            ) {
                Text(
                    text = "Elevated Card (Clickable)",
                    modifier = Modifier.padding(AtomicDesignSystem.spacing.MD),
                    style = AtomicDesignSystem.typography.BodyLarge
                )
            }
            
            Spacer(modifier = Modifier.height(AtomicDesignSystem.spacing.SM))
            
            // Molecule Forms Section
            Text(
                text = "Form Molecules",
                style = AtomicDesignSystem.typography.HeadlineSmall
            )
            
            SearchForm(
                searchQuery = searchQuery,
                onSearchQueryChange = { searchQuery = it },
                onSearch = { /* Search action */ }
            )
            
            Spacer(modifier = Modifier.height(AtomicDesignSystem.spacing.SM))
            
            SettingsForm(
                settingsData = settingsData,
                onSettingsChange = { settingsData = it }
            )
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun AtomicComponentShowcasePreview() {
    AtomicTheme {
        AtomicComponentShowcase()
    }
}

File: iosApp/iosApp/Views/Templates/AtomicComponentShowcase.swift
import SwiftUI

// MARK: - Atomic Component Showcase
// Template demonstrating all iOS atomic components and molecules
// Matches Android AtomicComponentShowcase.kt implementation

struct AtomicComponentShowcase: View {
    
    // MARK: - State Properties
    @State private var searchQuery = ""
    @State private var textFieldValue = ""
    @State private var isToggleOn = true
    @State private var isCheckboxChecked = false
    @State private var selectedTab = "home"
    @State private var selectedNavItem = "home"
    @State private var settingsData = SearchForm.SettingsData()
    @State private var email = ""
    @State private var password = ""
    @State private var showingSearchLoading = false
    @State private var showingLoginLoading = false
    
    // MARK: - Body
    var body: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: AtomicSpacing.lg) {
                    // Header
                    sectionHeader("iOS Atomic Design System")
                    
                    // Input Atoms Section
                    atomicInputsSection
                    
                    // Button Atoms Section
                    atomicButtonsSection
                    
                    // Card Atoms Section
                    atomicCardsSection
                    
                    // Navigation Atoms Section
                    atomicNavigationSection
                    
                    // Form Molecules Section
                    formMoleculesSection
                    
                    Spacer(minLength: AtomicSpacing.xl)
                }
                .atomicPadding()
            }
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .principal) {
                    AtomicNavigationBar(
                        title: "Atomic Components",
                        trailingActions: [
                            AtomicNavigationBar.NavigationAction(
                                icon: "info.circle",
                                accessibilityLabel: "Info"
                            ) { print("Info tapped") }
                        ],
                        centered: true
                    )
                }
            }
        }
    }
    
    // MARK: - Section Views
    
    private var atomicInputsSection: some View {
        VStack(alignment: .leading, spacing: AtomicSpacing.md) {
            sectionHeader("Input Atoms")
            
            AtomicCard {
                VStack(spacing: AtomicSpacing.md) {
                    AtomicTextField(
                        title: "Text Field",
                        placeholder: "Enter some text...",
                        text: $textFieldValue
                    )
                    
                    AtomicSearchField(
                        placeholder: "Search locations...",
                        text: $searchQuery
                    )
                    
                    AtomicToggle(
                        label: "Enable notifications",
                        isOn: $isToggleOn
                    )
                    
                    AtomicCheckbox(
                        label: "Accept terms and conditions",
                        isChecked: $isCheckboxChecked
                    )
                }
                .atomicPadding()
            }
        }
    }
    
    private var atomicButtonsSection: some View {
        VStack(alignment: .leading, spacing: AtomicSpacing.md) {
            sectionHeader("Button Atoms")
            
            AtomicCard {
                VStack(spacing: AtomicSpacing.sm) {
                    // Primary Button (using existing PrimaryButton from Buttons folder)
                    HStack {
                        Text("Primary Action")
                            .font(.system(size: 16, weight: .medium))
                            .foregroundColor(.atomicOnPrimary)
                        Spacer()
                    }
                    .frame(height: AtomicSpacing.buttonHeight)
                    .atomicPaddingMD(.horizontal)
                    .background(Color.atomicPrimary)
                    .cornerRadius(AtomicShapes.button)
                    .onTapGesture {
                        print("Primary button tapped")
                    }
                    
                    // Secondary Button
                    HStack {
                        Text("Secondary Action")
                            .font(.system(size: 16, weight: .medium))
                            .foregroundColor(.atomicPrimary)
                        Spacer()
                    }
                    .frame(height: AtomicSpacing.buttonHeight)
                    .atomicPaddingMD(.horizontal)
                    .background(Color.clear)
                    .overlay(
                        RoundedRectangle(cornerRadius: AtomicShapes.button)
                            .stroke(Color.atomicPrimary, lineWidth: AtomicShapes.borderMedium)
                    )
                    .onTapGesture {
                        print("Secondary button tapped")
                    }
                }
                .atomicPadding()
            }
        }
    }
    
    private var atomicCardsSection: some View {
        VStack(alignment: .leading, spacing: AtomicSpacing.md) {
            sectionHeader("Card Atoms")
            
            VStack(spacing: AtomicSpacing.sm) {
                AtomicCard(onClick: {
                    print("Basic card tapped")
                }) {
                    Text("Basic Card")
                        .font(.system(size: 16))
                        .atomicPadding()
                }
                
                AtomicOutlinedCard(onClick: {
                    print("Outlined card tapped")
                }) {
                    VStack(alignment: .leading, spacing: AtomicSpacing.xs) {
                        Text("Outlined Card")
                            .font(.system(size: 16, weight: .medium))
                        Text("This card has a border")
                            .font(.system(size: 14))
                            .foregroundColor(.atomicOnSurfaceVariant)
                    }
                    .atomicPadding()
                }
                
                AtomicElevatedCard(onClick: {
                    print("Elevated card tapped")
                }) {
                    HStack {
                        Image(systemName: "star.fill")
                            .foregroundColor(.atomicPrimary)
                            .font(.title2)
                        VStack(alignment: .leading, spacing: AtomicSpacing.xs) {
                            Text("Elevated Card")
                                .font(.system(size: 16, weight: .medium))
                            Text("Clickable with elevation")
                                .font(.system(size: 14))
                                .foregroundColor(.atomicOnSurfaceVariant)
                        }
                        Spacer()
                    }
                    .atomicPadding()
                }
                
                AtomicWeatherCard(
                    weatherCondition: .clear,
                    onClick: {
                        print("Weather card tapped")
                    }
                ) {
                    HStack {
                        Image(systemName: "sun.max.fill")
                            .foregroundColor(.atomicSunny)
                            .font(.title)
                        VStack(alignment: .leading, spacing: AtomicSpacing.xs) {
                            Text("Weather Card")
                                .font(.system(size: 16, weight: .medium))
                            Text("Sunny conditions")
                                .font(.system(size: 14))
                                .foregroundColor(.atomicOnSurfaceVariant)
                        }
                        Spacer()
                    }
                    .atomicPadding()
                }
            }
        }
    }
    
    private var atomicNavigationSection: some View {
        VStack(alignment: .leading, spacing: AtomicSpacing.md) {
            sectionHeader("Navigation Atoms")
            
            AtomicCard {
                VStack(spacing: AtomicSpacing.md) {
                    // Tab Bar Preview
                    AtomicTabBar(
                        selectedTab: $selectedTab,
                        tabs: [
                            AtomicTabBar.TabItem(id: "home", title: "Home", icon: "house"),
                            AtomicTabBar.TabItem(id: "favorites", title: "Favorites", icon: "heart", badgeCount: 3),
                            AtomicTabBar.TabItem(id: "settings", title: "Settings", icon: "gear")
                        ]
                    )
                    
                    Divider()
                    
                    // Navigation Rail Preview (Horizontal)
                    HStack(spacing: AtomicSpacing.sm) {
                        ForEach(["home", "search", "profile"], id: \.self) { item in
                            Button(action: {
                                selectedNavItem = item
                            }) {
                                VStack(spacing: AtomicSpacing.xs) {
                                    Image(systemName: iconForNavItem(item))
                                        .font(.system(size: 20, weight: selectedNavItem == item ? .semibold : .medium))
                                        .foregroundColor(selectedNavItem == item ? .atomicPrimary : .atomicOnSurfaceVariant)
                                    
                                    Text(item.capitalized)
                                        .font(.system(size: 12, weight: .medium))
                                        .foregroundColor(selectedNavItem == item ? .atomicPrimary : .atomicOnSurfaceVariant)
                                }
                                .frame(width: 60, height: 50)
                                .background(
                                    selectedNavItem == item ?
                                        Color.atomicPrimary.opacity(0.12) :
                                        Color.clear
                                )
                                .cornerRadius(AtomicShapes.sm)
                            }
                            .buttonStyle(PlainButtonStyle())
                        }
                    }
                    .frame(maxWidth: .infinity)
                }
                .atomicPadding()
            }
        }
    }
    
    private var formMoleculesSection: some View {
        VStack(alignment: .leading, spacing: AtomicSpacing.md) {
            sectionHeader("Form Molecules")
            
            VStack(spacing: AtomicSpacing.md) {
                // Search Form
                SearchForm(
                    searchQuery: $searchQuery,
                    onSearch: {
                        showingSearchLoading = true
                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                            showingSearchLoading = false
                        }
                    },
                    isLoading: showingSearchLoading
                )
                
                // Settings Form
                SettingsForm(settingsData: $settingsData)
                
                // Login Form
                LoginForm(
                    email: $email,
                    password: $password,
                    onLogin: {
                        showingLoginLoading = true
                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                            showingLoginLoading = false
                        }
                    },
                    onForgotPassword: {
                        print("Forgot password tapped")
                    },
                    isLoading: showingLoginLoading
                )
            }
        }
    }
    
    // MARK: - Helper Methods
    
    private func sectionHeader(_ title: String) -> some View {
        Text(title)
            .font(.system(size: 20, weight: .bold))
            .foregroundColor(.atomicOnSurface)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
    
    private func iconForNavItem(_ item: String) -> String {
        switch item {
        case "home":
            return "house"
        case "search":
            return "magnifyingglass"
        case "profile":
            return "person"
        default:
            return "circle"
        }
    }
}

// MARK: - Preview
struct AtomicComponentShowcase_Previews: PreviewProvider {
    static var previews: some View {
        AtomicComponentShowcase()
            .previewDevice("iPhone 14")
    }
}